recipes/newrelic/infrastructure/ohi/mongodb/debian.yml

```diff
@@ -62,24 +59,344 @@ preInstall:
+          if [ $IS_INFRA_INSTALLED -eq 0 ] ; then
+            echo "The infrastructure agent is required to install this integration, we recommend going through our guided install path for this pre-requisite which can be found here:  https://docs.newrelic.com/docs/full-stack-observability/observe-everything/get-started/new-relic-guided-install-overview" >&2
+            exit 1
+          fi
+        - |
+          # Map of tool names to the associated error code
+          REQUIRED_TOOLS_AND_ERROR_CODES="date:41 sha256sum:42 base64:43 head:44"
+
+          for tuple in $REQUIRED_TOOLS_AND_ERROR_CODES; do
+            tool=$(echo ${tuple} |cut -d':' -f1)
+            code=$(echo ${tuple} |cut -d':' -f2)
+
+            IS_TOOL_INSTALLED=$(which ${tool} | wc -l)
+            if [ "$IS_TOOL_INSTALLED" -eq 0 ]
+            then
+              echo -e "{{.RED}}This installation requires '${tool}' to be installed.{{.NOCOLOR}}" >> /dev/stderr
+              exit ${code}
+            fi
+          done
+        - | 
+          IS_MONGO_SHELL_INSTALLED=$(which mongo | wc -l)
+          if [ $IS_MONGO_SHELL_INSTALLED -eq 0 ]; then
+            echo "Mongo shell is required to run the newrelic install" # TODO: Prompt to install if not present?
+            exit 46
+          fi
+
+    mongo_pre_req:
+      cmds:
+        - |
+          if [ -f /tmp/mongo-role.js ]; then
+            rm -f /tmp/mongo-role.js 
+          fi
+          tee -a /tmp/mongo-role.js > /dev/null <<-"EOT"
+            use admin
+            db.createRole({ role: "listCollections", privileges: [{ resource: { db: "", collection: "" }, actions: [ "listCollections" ] }], roles: [] })
+          EOT
+          if [ -f /tmp/mongo-user.js ]; then
+            rm -f /tmp/mongo-user.js 
+          fi
+          tee -a /tmp/mongo-user.js > /dev/null <<-"EOT"
+            use admin
+            db.createUser({ user: "MONGO_MONITORING_USERNAME", pwd: "MONGO_MONITORING_PASSWORD", roles: [ "clusterMonitor", "listCollections" ] })
+          EOT
+      
+    assert_input:
+      cmds:
+        - |
+          NEW_RELIC_ASSUME_YES="{{.NEW_RELIC_ASSUME_YES}}" # TODO: Implement NEW_RELIC_ASSUME_YES
+          NR_CLI_DB_HOSTNAME="{{.NR_CLI_DB_HOSTNAME}}"
+          NR_CLI_DB_PORT="{{.NR_CLI_DB_PORT}}"
+          NR_CLI_DB_USERNAME="{{.NR_CLI_DB_USERNAME}}"
+          NR_CLI_DB_PASSWORD="{{.NR_CLI_DB_PASSWORD}}"
+          NR_CLI_DB_CLUSTERNAME="{{.NR_CLI_DB_CLUSTERNAME}}"
+          NR_CLI_DB_AUTH="{{.NR_CLI_DB_AUTH}}"
+          NR_CLI_SSL="{{.NR_CLI_SSL}}"
+          NR_CLI_CLIENT_CERT_FILE="{{.NR_CLI_CLIENT_CERT_FILE}}"
+          NR_CLI_CERT_AUTH_FILE="{{.NR_CLI_CERT_AUTH_FILE}}"
+
+          # MongoDB defaults
+          NR_CLI_DB_HOSTNAME=${NR_CLI_DB_HOSTNAME:-localhost}
+          NR_CLI_DB_PORT=${NR_CLI_DB_PORT:-27017}
+          NR_CLI_DB_CLUSTERNAME=${NR_CLI_DB_CLUSTERNAME:-mongocluster}
+          NR_CLI_DB_AUTH=${NR_CLI_DB_AUTH:-admin}
+
+          MONGO_MONITORING_USERNAME=newrelic
```

Comment:
env var should be configurable from env var too

Reply:
This was corrected.